drivers/src/pci/capabilities.rs

```diff
@@ -36,11 +38,94 @@ impl CapabilityId {
     }
 }
 
-// Maps a PCI capability ID to its offset in config space.
-#[derive(Clone, Copy, Debug)]
+mod header_offsets {
+    use super::CapabilityHeader;
+    use crate::define_field_span;
+
+    define_field_span!(CapabilityHeader, cap_id, u8);
+    define_field_span!(CapabilityHeader, next_cap, u8);
+}
+
+// Represents a single PCI capability.
 struct PciCapability {
     id: CapabilityId,
     offset: usize,
+    next: usize,
+}
+
+impl PciCapability {
+    // Creates a new capability of type `id` at `ptr`, which itself is at `offset` within the
+    // configuration space.
+    //
+    // # Safety
+    //
+    // `ptr` must refer to a uniquely owned PCI capability structure of the specified type.
+    unsafe fn new(
+        _ptr: NonNull<CapabilityHeader>,
```

Comment:
Similar question about passing a reference safely instead of a pointer.

Reply:
Done.